src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,723 @@
+        stored. Default: os.path.curdir.
+    ignore_warnings : bool
+        Turn all errors into warnings.
+
+    Attributes
+    ----------
+    logger : Logger
+        The logging object.
+    file_list : list
+        The list of filenames sorted by the lambda state.
+    '''
+    def __init__(self, T, units='kT', software='GROMACS', dir=os.path.curdir,
+                 prefix='dhdl', suffix='xvg',
+                 outdirectory=os.path.curdir,
+                 ignore_warnings=False):
+
+        super().__init__(units, software, T, outdirectory)
+        self.ignore_warnings = ignore_warnings
+        self.logger = logging.getLogger('alchemlyb.workflows.ABFE')
+        self.logger.info('Initialise Alchemlyb ABFE Workflow')
+        self.logger.info(f'Alchemlyb Version: f{__version__}')
+        self.logger.info(f'Set Temperature to {T} K.')
+        self.logger.info(f'Set Software to {software}.')
+
+        self.update_units(units)
+
+        self.logger.info(f'Finding files with prefix: {prefix}, suffix: '
+                         f'{suffix} under directory {dir} produced by '
+                         f'{software}')
+        self.file_list = glob(dir + '/**/' + prefix + '*' + suffix,
+                         recursive=True)
+
+        self.logger.info(f'Found {len(self.file_list)} xvg files.')
+        self.logger.info("Unsorted file list: \n %s", '\n'.join(
+            self.file_list))
+
+        if software == 'GROMACS':
+            self.logger.info(f'Using {software} parser to read the data.')
+            self._extract_u_nk = gmx.extract_u_nk
+            self._extract_dHdl = gmx.extract_dHdl
+        elif software == 'AMBER':
+            self._extract_u_nk = amber.extract_u_nk
+            self._extract_dHdl = amber.extract_dHdl
+        else:
+            raise NotImplementedError(f'{software} parser not found.')
+
+    def read(self):
+        '''Read the u_nk and dHdL data from the
+        :attr:`~alchemlyb.workflows.ABFE.file_list`
+
+        Attributes
+        ----------
+        u_nk_list : list
+            A list of :class:`pandas.DataFrame` of u_nk.
+        dHdl_list : list
+            A list of :class:`pandas.DataFrame` of dHdl.
+        '''
+        u_nk_list = []
+        dHdl_list = []
+        for file in self.file_list:
+            try:
+                u_nk = self._extract_u_nk(file, T=self.T)
+                self.logger.info(
+                    f'Reading {len(u_nk)} lines of u_nk from {file}')
+                u_nk_list.append(u_nk)
+            except Exception as exc:
+                msg = f'Error reading read u_nk from {file}.'
+                if self.ignore_warnings:
+                    self.logger.exception(msg + f'\n{exc}\n' +
+                        'This exception is being ignored because ignore_warnings=True.')
+                else:
+                    self.logger.error(msg)
+                    raise exc from ValueError(msg)
```

Comment:
Ok, `raise exc from ValueError(msg)` will raise the original error. It will look a bit funny, though, because it will say that the original error was caused by the ValueError that we introduced purely for the sake of adding the information about the file. I read a bit more about exception chaining https://stackoverflow.com/a/24752607/334357 and https://peps.python.org/pep-3134/#explicit-exception-chaining and that changed my mind because I learned that it seems normal to have an API error (currently ValueError) and then ask for the cause in the `apierror.__cause__` attribute.

I still don't like ValueError but I concede that your original solution
```python
raise ValueError from exc
```
was the right approach.

I'd like to raise an `OSError` as our generic error:
```suggestion
                    raise OSError(msg) from exc
```
This is appropriate for I/O and then the cause will make more sense.

Reply:
Is this exception tested anywhere??